avm/res/document-db/database-account/main.bicep

```diff
@@ -614,3 +693,54 @@ type sqlDatabaseContainer = {
     paths: string[]
   }[]?
 }
+
+type secretsKeyVault = {
+  @description('Required. The key vault name where to store the keys and connection strings generated by the modules.')
+  keyVaultName: string
+
+  @description('Optional. Default to the resource group where this account is. The resource group name where the key vault is.')
+  resourceGroupName: string?
+
+  @description('Optional. Default to Primary-Write-Key. The primary write key secret name to create.')
+  primaryWriteKeySecretName: string?
+
+  @description('Optional. Default to Primary-Readonly-Key. The primary readonly key secret name to create.')
+  primaryReadOnlyKeySecretName: string?
+
+  @description('Optional. Default to Primary-Write-ConnectionString. The primary write connection string secret name to create.')
+  primaryWriteConnectionStringSecretName: string?
+
+  @description('Optional. Default to Primary-Readonly-ConnectionString. The primary readonly connection string secret name to create.')
+  primaryReadonlyConnectionStringSecretName: string?
+
+  @description('Optional. Default to Secondary-Write-Key. The primary write key secret name to create.')
+  secondaryWriteKeySecretName: string?
+
+  @description('Optional. Default to Secondary-Readonly-Key. The primary readonly key secret name to create.')
+  secondaryReadonlyKeySecretName: string?
+
+  @description('Optional. Default to Secondary-Write-ConnectionString. The primary write connection string secret name to create.')
+  secondaryWriteConnectionStringSecretName: string?
+
+  @description('Optional. Default to Secondary-Readonly-ConnectionString. The primary readonly connection string secret name to create.')
+  secondaryReadonlyConnectionStringSecretName: string?
+}
+
+type networkRestrictionsConfig = {
```

Comment:
Just a suggestion, but I'd recommend to add a suffix `Type` at the end of the User-defined-Types in a consistent way as it makes naming easier. For example the `networkRestrictions` parameter could just have a `networkRestrictionsType`.

Just a suggestion, so feel free to not do this if you prefer the current implementation

Reply:
Added the suffixes